src/parser/jobs/rpr/modules/Gauge.tsx

```diff
@@ -0,0 +1,178 @@
+import {Trans} from '@lingui/react'
+import Color from 'color'
+import {DataLink} from 'components/ui/DbLink'
+import {ActionKey} from 'data/ACTIONS'
+import {Event} from 'event'
+import {filter} from 'parser/core/filter'
+import {dependency} from 'parser/core/Injectable'
+import {CounterGauge, Gauge as CoreGauge} from 'parser/core/modules/Gauge'
+import Suggestions, {SEVERITY, TieredSuggestion} from 'parser/core/modules/Suggestions'
+import React from 'react'
+
+const SOUL_MAX_VALUE = 100
+const SHROUD_MAX_VALUE = 100
+
+const SOUL_GAUGE_COLOR = Color('#DC0707FF')
+const SHROUD_GAUGE_COLOR = Color('#2C9FCB')
+
+const SOUL_GENERATORS: ActionKey[] = [
+	'SLICE',
+	'WAXING_SLICE',
+	'INFERNAL_SLICE',
+	'SPINNING_SCYTHE',
+	'NIGHTMARE_SCYTHE',
+]
+
+/**
+ * Actions on cooldown that generate souls, those generate
+ * more souls than regular combo actions
+ */
+const SOUL_GENERATORS_CDS: ActionKey[] = [
+	'SOUL_SLICE',
+	'SOUL_SCYTHE',
+]
+
+const SOUL_CONSUMERS: ActionKey[] = [
+	'BLOOD_STALK',
+	'GRIM_SWATHE',
+	'GLUTTONY',
+	'UNVEILED_GIBBET',
+	'UNVEILED_GALLOWS',
+]
+
+const SHROUD_GENERATORS: ActionKey[] = [
+	'GIBBET',
+	'GALLOWS',
+	'GUILLOTINE',
+]
+
+const SHROUD_CONSUMERS: ActionKey[] = [
+	'ENSHROUD',
+]
+
+/**
+ * Base amount for the combo actions
+ */
+const BASE_GAUGE_GENERATION_AMOUNT = 10
+
+/**
+ * Base amount for the soul gauge consumption
+ */
+const BASE_GAUGE_CONSUMPTION_AMOUNT = 50
+const PLENTIFUL_HARVEST_SHROUD_GENERATION_AMOUNT = 50
+const ENSHROUD_GAUGE_CONSUMPTION = 50
+
+/**
+ * Amount generated for the "Soul" skills, which are cooldowns
+ */
+const SOUL_GAUGE_GENERATION_AMOUNT = 50
```

Comment:
Unnecessarily large comments, these can be reordered for sanity imo. Harvest is also clearly for shroud, so it doesn't really need it in the name. Not sure why Enshroud needs a special const either.